support/cas-server-support-oauth-core-api/src/main/java/org/apereo/cas/support/oauth/web/endpoints/OAuth20RevocationEndpointController.java

```diff
@@ -0,0 +1,194 @@
+package org.apereo.cas.support.oauth.web.endpoints;
+
+import org.apereo.cas.audit.AuditableContext;
+import org.apereo.cas.support.oauth.OAuth20Constants;
+import org.apereo.cas.support.oauth.services.OAuthRegisteredService;
+import org.apereo.cas.support.oauth.util.OAuth20Utils;
+import org.apereo.cas.ticket.OAuth20Token;
+import org.apereo.cas.ticket.accesstoken.OAuth20AccessToken;
+import org.apereo.cas.ticket.refreshtoken.OAuth20RefreshToken;
+
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.apache.commons.lang3.StringUtils;
+import org.pac4j.core.context.JEEContext;
+import org.pac4j.core.profile.CommonProfile;
+import org.pac4j.core.profile.ProfileManager;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.servlet.ModelAndView;
+import org.springframework.web.servlet.view.json.MappingJackson2JsonView;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * This is {@link OAuth20RevocationEndpointController}.
+ *
+ * @author Julien Huon
+ * @since 6.2.0
+ */
+@Slf4j
+public class OAuth20RevocationEndpointController extends BaseOAuth20Controller {
+    public OAuth20RevocationEndpointController(final OAuth20ConfigurationContext oAuthConfigurationContext) {
+        super(oAuthConfigurationContext);
+    }
+
+    /**
+     * Handle request for revocation.
+     *
+     * @param request  the request
+     * @param response the response
+     * @return the response entity
+     */
+    @PostMapping(path = '/' + OAuth20Constants.BASE_OAUTH20_URL + '/' + OAuth20Constants.REVOCATION_URL,
+        produces = MediaType.APPLICATION_JSON_VALUE)
+    public ModelAndView handleRequest(final HttpServletRequest request,
+                                      final HttpServletResponse response) {
+        val context = new JEEContext(request, response, getOAuthConfigurationContext().getSessionStore());
+
+        if (!verifyRevocationRequest(context)) {
+            LOGGER.error("Revocation request verification failed. Request is missing required parameters");
```

Comment:
This likely should be a warn.

Reply:
Are you sure?

All the other endpoints are raising an error when the request verification failed. For example:

Authorize Endpoint:
```
if (!verifyAuthorizeRequest(context) || !isRequestAuthenticated(manager)) {
            LOGGER.error("Authorize request verification failed. Authorization request is missing required parameters, "
                + "or the request is not authenticated and contains no authenticated profile/principal.");
            return OAuth20Utils.produceUnauthorizedErrorView();
        }
```

AccessToken Endpoint:

```
            if (!verifyAccessTokenRequest(request, response)) {
                throw new IllegalArgumentException("Access token validation failed");
            }
```

Reply:
Sounds good, let's keep things consistent then. Thanks!